[2025-01-16T08:21:40.355+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-01-16T08:21:40.401+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ETL.extract_data manual__2025-01-16T08:21:33.379235+00:00 [queued]>
[2025-01-16T08:21:40.418+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ETL.extract_data manual__2025-01-16T08:21:33.379235+00:00 [queued]>
[2025-01-16T08:21:40.420+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-01-16T08:21:40.445+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): extract_data> on 2025-01-16 08:21:33.379235+00:00
[2025-01-16T08:21:40.451+0000] {standard_task_runner.py:63} INFO - Started process 2216 to run task
[2025-01-16T08:21:40.451+0000] {clientserver.py:543} INFO - Closing down clientserver connection
[2025-01-16T08:21:40.460+0000] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'ETL', 'extract_data', 'manual__2025-01-16T08:21:33.379235+00:00', '--job-id', '29', '--raw', '--subdir', 'DAGS_FOLDER/ETL.py', '--cfg-path', '/tmp/tmp1ksyh7tp']
[2025-01-16T08:21:40.464+0000] {standard_task_runner.py:91} INFO - Job 29: Subtask extract_data
[2025-01-16T08:21:40.543+0000] {task_command.py:426} INFO - Running <TaskInstance: ETL.extract_data manual__2025-01-16T08:21:33.379235+00:00 [running]> on host airflow-scheduler
[2025-01-16T08:21:40.680+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Ajied(WhatsApp), Ajied(Email)' AIRFLOW_CTX_DAG_ID='ETL' AIRFLOW_CTX_TASK_ID='extract_data' AIRFLOW_CTX_EXECUTION_DATE='2025-01-16T08:21:33.379235+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-16T08:21:33.379235+00:00'
[2025-01-16T08:21:40.682+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-01-16T08:21:40.694+0000] {base.py:84} INFO - Using connection ID 'spark_main' for task execution.
[2025-01-16T08:21:40.696+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --jars /spark-scripts/jars/jars_postgresql-42.2.20.jar --name arrow-spark /spark-scripts/resources/extract.py
[2025-01-16T08:21:40.745+0000] {clientserver.py:505} INFO - Error while sending or receiving.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 503, in send_command
    self.socket.sendall(command.encode("utf-8"))
ConnectionResetError: [Errno 104] Connection reset by peer
[2025-01-16T08:21:40.747+0000] {clientserver.py:543} INFO - Closing down clientserver connection
[2025-01-16T08:21:40.748+0000] {java_gateway.py:1052} INFO - Exception while sending command.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 503, in send_command
    self.socket.sendall(command.encode("utf-8"))
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1038, in send_command
    response = connection.send_command(command)
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 506, in send_command
    raise Py4JNetworkError(
py4j.protocol.Py4JNetworkError: Error while sending
[2025-01-16T08:21:40.753+0000] {clientserver.py:543} INFO - Closing down clientserver connection
[2025-01-16T08:21:42.700+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:42 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-01-16T08:21:44.290+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkContext: Running Spark version 3.3.2
[2025-01-16T08:21:44.321+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceUtils: ==============================================================
[2025-01-16T08:21:44.322+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-01-16T08:21:44.323+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceUtils: ==============================================================
[2025-01-16T08:21:44.324+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkContext: Submitted application: Data Extract
[2025-01-16T08:21:44.347+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-01-16T08:21:44.357+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceProfile: Limiting resource is cpu
[2025-01-16T08:21:44.358+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-01-16T08:21:44.407+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SecurityManager: Changing view acls to: airflow
[2025-01-16T08:21:44.408+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SecurityManager: Changing modify acls to: airflow
[2025-01-16T08:21:44.408+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SecurityManager: Changing view acls groups to:
[2025-01-16T08:21:44.409+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SecurityManager: Changing modify acls groups to:
[2025-01-16T08:21:44.410+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(airflow); groups with view permissions: Set(); users  with modify permissions: Set(airflow); groups with modify permissions: Set()
[2025-01-16T08:21:44.632+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO Utils: Successfully started service 'sparkDriver' on port 36149.
[2025-01-16T08:21:44.662+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkEnv: Registering MapOutputTracker
[2025-01-16T08:21:44.706+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkEnv: Registering BlockManagerMaster
[2025-01-16T08:21:44.729+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-01-16T08:21:44.730+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-01-16T08:21:44.735+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-01-16T08:21:44.761+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-2e071a17-2faf-4e24-9d3b-4d3812e5c671
[2025-01-16T08:21:44.780+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2025-01-16T08:21:44.805+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-01-16T08:21:44.980+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-01-16T08:21:44.988+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:44 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2025-01-16T08:21:45.032+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 ERROR SparkContext: Failed to add /opt/bitnami/spark/jars/postgresql-42.2.18.jar to Spark environment
[2025-01-16T08:21:45.033+0000] {spark_submit.py:641} INFO - java.io.FileNotFoundException: Jar /opt/bitnami/spark/jars/postgresql-42.2.18.jar not found
[2025-01-16T08:21:45.034+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addLocalJarFile$1(SparkContext.scala:1959)
[2025-01-16T08:21:45.034+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.addJar(SparkContext.scala:2014)
[2025-01-16T08:21:45.035+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$12(SparkContext.scala:507)
[2025-01-16T08:21:45.036+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.$anonfun$new$12$adapted(SparkContext.scala:507)
[2025-01-16T08:21:45.036+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-01-16T08:21:45.037+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-01-16T08:21:45.038+0000] {spark_submit.py:641} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-01-16T08:21:45.038+0000] {spark_submit.py:641} INFO - at org.apache.spark.SparkContext.<init>(SparkContext.scala:507)
[2025-01-16T08:21:45.039+0000] {spark_submit.py:641} INFO - at org.apache.spark.api.java.JavaSparkContext.<init>(JavaSparkContext.scala:58)
[2025-01-16T08:21:45.040+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-01-16T08:21:45.041+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-01-16T08:21:45.041+0000] {spark_submit.py:641} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-01-16T08:21:45.042+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-01-16T08:21:45.042+0000] {spark_submit.py:641} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-01-16T08:21:45.043+0000] {spark_submit.py:641} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2025-01-16T08:21:45.044+0000] {spark_submit.py:641} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2025-01-16T08:21:45.044+0000] {spark_submit.py:641} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2025-01-16T08:21:45.045+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2025-01-16T08:21:45.046+0000] {spark_submit.py:641} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2025-01-16T08:21:45.046+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-01-16T08:21:45.047+0000] {spark_submit.py:641} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-01-16T08:21:45.047+0000] {spark_submit.py:641} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-01-16T08:21:45.118+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2025-01-16T08:21:45.168+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.5:7077 after 28 ms (0 ms spent in bootstraps)
[2025-01-16T08:21:45.266+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20250116082145-0026
[2025-01-16T08:21:45.273+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 43813.
[2025-01-16T08:21:45.274+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO NettyBlockTransferService: Server created on airflow-scheduler:43813
[2025-01-16T08:21:45.275+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-01-16T08:21:45.281+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, airflow-scheduler, 43813, None)
[2025-01-16T08:21:45.285+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO BlockManagerMasterEndpoint: Registering block manager airflow-scheduler:43813 with 434.4 MiB RAM, BlockManagerId(driver, airflow-scheduler, 43813, None)
[2025-01-16T08:21:45.287+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, airflow-scheduler, 43813, None)
[2025-01-16T08:21:45.289+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, airflow-scheduler, 43813, None)
[2025-01-16T08:21:45.480+0000] {spark_submit.py:641} INFO - 25/01/16 08:21:45 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2025-01-16T08:21:45.638+0000] {spark_submit.py:641} INFO - Extracting data from PostgreSQL (products)...
[2025-01-16T08:21:48.136+0000] {spark_submit.py:641} INFO - Extracting data from PostgreSQL (transactions)...
[2025-01-16T08:21:48.170+0000] {spark_submit.py:641} INFO - Products Data:
[2025-01-16T08:22:04.442+0000] {spark_submit.py:641} INFO - 25/01/16 08:22:04 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-16T08:22:19.440+0000] {spark_submit.py:641} INFO - 25/01/16 08:22:19 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-16T08:22:34.441+0000] {spark_submit.py:641} INFO - 25/01/16 08:22:34 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
[2025-01-16T08:22:40.682+0000] {timeout.py:68} ERROR - Process timed out, PID: 2216
[2025-01-16T08:22:40.685+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-01-16T08:22:40.686+0000] {spark_submit.py:767} INFO - Sending kill signal to spark-submit
[2025-01-16T08:22:40.721+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 460, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 177, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 555, in submit
    self._process_spark_submit_log(iter(self._submit_sp.stdout))  # type: ignore
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 604, in _process_spark_submit_log
    for line in itr:
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/timeout.py", line 69, in handle_timeout
    raise AirflowTaskTimeout(self.error_message)
airflow.exceptions.AirflowTaskTimeout: Timeout, PID: 2216
[2025-01-16T08:22:40.742+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=ETL, task_id=extract_data, run_id=manual__2025-01-16T08:21:33.379235+00:00, execution_date=20250116T082133, start_date=20250116T082140, end_date=20250116T082240
[2025-01-16T08:22:40.839+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 2
[2025-01-16T08:22:40.870+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
